Sources/Apodini/Semantic Model Builder/InterfaceExporter/EnrichedContent.swift

```diff
@@ -33,48 +32,75 @@ public struct EnrichedContent: Encodable {
+            .formatRelationships(into: initialValue, with: formatter, context: context)
+    }
 
-        return destinations.formatRelationships(into: initialValue, with: formatter, context: context)
+    /// Used to apply a `RelationshipFormatter` to all Relationships of the `Endpoint`
+    /// with the context of this `EnrichedContent` (path parameter values and property values of the response).
+    ///
+    /// - Parameters:
+    ///   - initialValue: The initial value the `RelationshipFormatter` should reduce into.
+    ///   - formatter: The actual instance of the `RelationshipFormatter`.
+    /// - Returns: The formatted result.
+    public func formatRelationships<Formatter: RelationshipFormatter>(
+        into initialValue: Formatter.Result,
+        with formatter: Formatter
+    ) -> Formatter.Result {
+        let context = ResolveContext(content: response.wrappedValue, parameters: parameters)
+
+        return endpoint
+            .relationships()
+            .formatRelationships(into: initialValue, with: formatter, context: context)
     }
 
-    /// Used to apply a `RelationshipFormatter` to the self relationships of the `Endpoint`
+    /// Used to apply a `RelationshipFormatter` to all Relationships of the `Endpoint`
     /// with the context of this `EnrichedContent` (path parameter values and property values of the response).
-    /// If nil is supplied for the `Operation`, the formatter is called for any `Operation`.
-    /// If there is no self relationship for a specified `Operation`, this call simply returns the initial value.
     ///
     /// - Parameters:
     ///   - initialValue: The initial value the `RelationshipFormatter` should reduce into.
     ///   - formatter: The actual instance of the `RelationshipFormatter`.
-    ///   - operation: The `Operation` for which to retrieve all self `RelationshipDestination` from the given `Endpoint`.
-    /// - Returns: The formatted result
-    func formatSelfRelationships<Formatter: RelationshipFormatter>(
+    ///   - sortKeyPath: The relationships will be applied to the formatter in the given order.
+    /// - Returns: The formatted result.
+    public func formatRelationships<Formatter: RelationshipFormatter, T: Comparable>(
         into initialValue: Formatter.Result,
         with formatter: Formatter,
-        for operation: Operation? = nil
+        sortedBy sortKeyPath: KeyPath<Operation, T>
     ) -> Formatter.Result {
         let context = ResolveContext(content: response.wrappedValue, parameters: parameters)
 
         var result = initialValue
 
-        if let specifiedOperation = operation {
-            endpoint.selfRelationship(for: specifiedOperation)?.formatRelationship(with: formatter, result: &result, context: context)
-        } else {
-            result = endpoint.selfRelationships().formatRelationships(into: result, with: formatter, context: context)
+        for operation in Operation.allCases.sorted(by: sortKeyPath) {
```

Comment:
You need to move `Collection+SortedByKeyPath.swift` back into the `Apodini` target 🙂

Reply:
Yeah saw it, but couldn’t yet commit it.
Is it okay to make that public? We would need to as it is used in the ProtobufferExporter.